File: src/java/jclongra/core/Primitives.java
/*
 * Copyright (c) Konrad Grzanek. All rights reserved.
 * Created 2016-03-20
 */
package jclongra.core;

import clojure.lang.RT;

public final class Primitives {

  public static boolean booleanNot(boolean b) {
    return !b;
  }

  public static long[] makeLongs(long size) {
    return new long[RT.intCast(size)];
  }

  public static double[] makeDoubles(long size) {
    return new double[RT.intCast(size)];
  }

}

File: src/clj/clongra/core/primitives.clj
;; Copyright (c) Konrad Grzanek. All rights reserved.
;; Created 2014-03-20

(in-ns 'clongra.core)

(defmacro boolean-not [b] `(jclongra.core.Primitives/booleanNot ~b))

(defn make-longs ;:- long -> long[]
  {:inline (fn [size] `(jclongra.core.Primitives/makeLongs ~size))}
  [^long size]
  (jclongra.core.Primitives/makeLongs size))


(defn make-doubles ;:- long -> double[]
  {:inline (fn [size] `(jclongra.core.Primitives/makeDoubles ~size))}
  [^long size]
  (jclongra.core.Primitives/makeDoubles size))
